src/pages/Graph/GraphToolbar/GraphSettings.tsx

```diff
@@ -824,6 +833,14 @@ class GraphSettings extends React.PureComponent<GraphSettingsProps, GraphSetting
     const newEdgeLabels = this.props.edgeLabels.filter(l => !isThroughputMode(l));
     this.props.setEdgeLabels([...newEdgeLabels, EdgeLabelMode.THROUGHPUT_GROUP, mode]);
   };
+
+  private toggleRankByMode = (mode: RankMode) => {
+    if (this.props.rankBy.includes(mode)) {
+      this.props.setRankBy(this.props.rankBy.filter(r => r !== mode));
+    } else {
+      this.props.setRankBy([...this.props.rankBy, mode]);
+    }
+  };
```

Comment:
My only motivation for having this logic in the reducer was to keep logic related to updating the redux state in the reducers but it did add another action and having this here makes it more consistent with the other toggles.

Reply:
OK, I wasn't sure about the original motivation.  I can understand it, and from what i can see we don't really do anything very useful in our reducers.  But in this case the change above seems straightforward and consistent with other handling in GraphSettings,  which is a plus.